### PR TITLE
[FEATURE] Add preview renderer

### PR DESCRIPTION
**Pull request by brotkrueml**
Related: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/10.3/Feature-78450-IntroducePreviewRendererPattern.html
Resolves: #781
Releases: main, 12.4, 11.5

### CODE REVIEW

File: Documentation/Ctrl/Properties/Index.rst [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
-   TransOrigDiffSourceField
-   TransOrigPointerField
-   Tstamp
-   Type
-   TypeiconClasses
-   TypeiconColumn
-   UseColumnsForDefaultValues
-   VersioningWS
-   VersioningWSAlwaysAllowLiveEdit
+    AdminOnly
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/Index.rst:10]: Can this not be simply solved by a glob?

File: Documentation/Ctrl/Properties/PreviewRenderer.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+previewRenderer
+===============
+
+.. confval:: previewRenderer
+
+   :Path: $GLOBALS['TCA'][$table]['ctrl']
+   :type: string
+   :Scope: Display
+
+   Configure a backend preview for a content element.
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/PreviewRenderer.rst:14]:
```suggestion
   Configures a backend preview for a content element.
```

File: Documentation/Ctrl/Properties/PreviewRenderer.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Examples
+========
+
+Have also a look at :ref:`t3coreapi:ConfigureCE-Preview` for more details.
+
+Use for any record in a table
+-----------------------------
+
+This specifies the preview renderer to be used for any record in :sql:`my_table`:
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/PreviewRenderer.rst:24]: should we use an allowed table name format such as tx_myextensions_table?

File: Documentation/Ctrl/Properties/PreviewRenderer.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+..  code-block:: php
+
+    $GLOBALS['TCA']['my_table']['ctrl']['previewRenderer']
+        = \MyVendor\MyExtension\Preview\PreviewRenderer::class;
+
+
+Table has a "type" field/attribute
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/PreviewRenderer.rst:32]: Why is the type in quotation marks?

File: Documentation/Ctrl/Properties/PreviewRenderer.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Have also a look at :ref:`t3coreapi:ConfigureCE-Preview` for more details.
+
+Use for any record in a table
+-----------------------------
+
+This specifies the preview renderer to be used for any record in :sql:`my_table`:
+
+..  code-block:: php
+
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/PreviewRenderer.rst:27]: Where would we put these examples? In the overrides? We should state so

File: Documentation/Ctrl/Properties/PreviewRenderer.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    $GLOBALS['TCA']['my_table']['types'][$type]['previewRenderer']
+        = \MyVendor\MyExtension\Preview\PreviewRenderer::class;
+
+Table has a "subtype_value_field" setting
+-----------------------------------------
+
+If your table and field have a :php:`subtype_value_field` TCA setting (like
```
Comment (linawolf) [inline Documentation/Ctrl/Properties/PreviewRenderer.rst:46]: Can we link this to further explainations?

**Review by linawolf — COMMENTED** [review]
Thanks for the work you put into this, just some small remarks

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 4 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst:32]: Maybe marking as sql is better.

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 5 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst:27]: The example as is works only in overrides, but you can also configure in the default table. So I stepped away from adding a caption.

File: Documentation/Ctrl/Properties/Index.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/Index.rst:10]: Done

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 3 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst]: Done

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 4 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst]: Done

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 5 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst:28]: I added a note.

File: Documentation/Ctrl/Properties/PreviewRenderer.rst (hunk 6 above)
Comment (brotkrueml) [reply Documentation/Ctrl/Properties/PreviewRenderer.rst:55]: Done

**Review by linawolf — APPROVED** [review]

**Pull request merged**